storage/paths/db_index.go

```diff
@@ -322,24 +321,31 @@ func (dbi *DBIndex) StorageDetach(ctx context.Context, id storiface.ID, url stri
 }
 
 func (dbi *DBIndex) StorageReportHealth(ctx context.Context, id storiface.ID, report storiface.HealthReport) error {
-
-	var canSeal, canStore bool
-	err := dbi.harmonyDB.QueryRow(ctx,
-		"SELECT can_seal, can_store FROM storage_path WHERE storage_id=$1", id).Scan(&canSeal, &canStore)
-	if err != nil {
-		return xerrors.Errorf("Querying for storage id %s fails with err %v", id, err)
-	}
-
-	_, err = dbi.harmonyDB.Exec(ctx,
-		"UPDATE storage_path set capacity=$1, available=$2, fs_available=$3, reserved=$4, used=$5, last_heartbeat=$6",
+	retryWait := time.Millisecond * 20
+retryReportHealth:
+	_, err := dbi.harmonyDB.Exec(ctx,
+		"UPDATE storage_path set capacity=$1, available=$2, fs_available=$3, reserved=$4, used=$5, last_heartbeat=NOW() where storage_id=$6",
 		report.Stat.Capacity,
 		report.Stat.Available,
 		report.Stat.FSAvailable,
 		report.Stat.Reserved,
 		report.Stat.Used,
-		time.Now())
+		id)
+	if err != nil {
+		//return xerrors.Errorf("updating storage health in DB fails with err: %v", err)
+		if harmonydb.IsErrSerialization(err) {
```

Comment:
is this replaced by the serialization option?

Reply:
Yes, I wrote this code before adding the option; Will change